sdcm/utils/url_utils.py

```diff
@@ -0,0 +1,87 @@
+import re
+from collections import namedtuple
+from urllib.request import urlopen
+
+from sdcm.utils.common import ParallelObject
+
+SCYLLA_URL_RESPONSE_TIMEOUT = 30
+SCYLLA_URL_REGEX = re.compile('.*(?P<url>http.*scylladb.com.*).*')
+SUPPORTED_FILE_EXTENSIONS = ('list', 'repo', 'xml', 'Packages', 'gz')
+
+BASE_ARCH_KEY = '$basearch'
+RELEASE_VERSION_KEY = '$releasever'
+URL_ENV_VARIABLES = {
+    BASE_ARCH_KEY: ['x86_64', 'noarch'],
+    RELEASE_VERSION_KEY: ['7'],
+}
+
+UrlDetails = namedtuple("UrlDetails", "is_xml_file urls")
+UrlDetails.__new__.__defaults__ = (None,) * len(UrlDetails._fields)
+
+
+def validate_url_is_file_and_extract_scylla_urls(urls, validate_file_output=True):
+    def _validate_url(_url):
+        is_xml_file, is_scylla_url = False, SCYLLA_URL_REGEX.match(_url) is not None
+
+        file_extension = _url.rsplit('/', maxsplit=1)[1]
+        if file_extension.endswith(('xml', 'xml.gz',)):
+            is_xml_file = True
+        if not file_extension.endswith(SUPPORTED_FILE_EXTENSIONS):
```

Comment:
remove this condition, we don't care the file extensions, we just read them.
and we shouldn't care about if it's a scylla domain or not